doc/classes/NavigationMeshGenerator.xml

```diff
@@ -14,12 +14,12 @@
 		<link title="Using NavigationMeshes">$DOCS_URL/tutorials/navigation/navigation_using_navigationmeshes.html</link>
 	</tutorials>
 	<methods>
-		<method name="bake" deprecated="">
+		<method name="bake" deprecated="This method is deprecated due to core threading changes. To upgrade existing code, first create a [NavigationMeshSourceGeometryData3D] resource. Use this resource with [method parse_source_geometry_data] to parse the [SceneTree] for nodes that should contribute to the navigation mesh baking. The [SceneTree] parsing needs to happen on the main thread. After the parsing is finished use the resource with [method bake_from_source_geometry_data] to bake a navigation mesh.">
```

Comment:
In my opinion, the message should be as short as possible, details should be placed in the description. In the future we might use deprecated/experimental messages for warnings in GDScript and C#.

Reply:
I do agree a lot, but I also wanted to keep my reasoning consistent in the PR. "Why", then "provide alternative". So moving it in the description would go against that.